app/main/posts/views/post-view-data.directive.js

```diff
@@ -99,49 +104,28 @@ function PostViewDataController(
                 }
             });
         }
-    });
+    }));
 
     // And this is for when you change the post status using the ... button
     // Because it's just the post status only, we're just checking if it matches
     // filters on the frontend. This makes it MUCH faster than using the API
-    $rootScope.$on('event:edit:post:status:data:mode:saveSuccess', function (event, args) {
+    unbindFns.push($rootScope.$on('event:edit:post:status:data:mode:saveSuccess', function (event, args) {
         let postObj = args.post;
         if (!newStatusMatchesFilters(postObj)) {
             removePostFromList(postObj);
         }
-    });
+    }));
 
-    function removePostFromList(postObj) {
-        $scope.posts.forEach((post, index) => {
-            // args.post is the post being updated/saved and sent from the broadcast
-            if (post.id === postObj.id) {
-                let nextInLine = $scope.posts[index + 1];
-                $scope.posts.splice(index, 1);
-                if ($scope.posts.length) {
-                    groupPosts($scope.posts);
-                    $scope.selectedPost.post = nextInLine;
-                    $scope.selectedPostId = nextInLine.id;
-                } else {
-                    $scope.selectedPost = {post: null, next: {}};
-                    $scope.selectedPostId = null;
-                    getPosts(false, false);
-                }
-            }
-        });
-    }
+    unbindFns.push($transitions.onSuccess({
+        to: 'posts.data.**'
+    }, () => {
+        $scope.activeCol = $state.params.activeCol;
+    }));
 
-    function newStatusMatchesFilters(postObj) {
-        let filters = $scope.hasFilters() ?  $scope.filters.status : PostFilters.getDefaults().status;
-        let matchingStatus = false;
-
-        filters.forEach((status) => {
-            if (postObj.status === status) {
-                matchingStatus = true;
-            }
-        });
-
-        return matchingStatus;
-    }
+    // Cleanup and remove all listeners
+    $scope.$on('$destroy', () => {
```

Comment:
Did some clean up while here to avoid leaking a bunch of event listeners everytime we go to data view